custom/icds_reports/migrations/sql_templates/database_views/service_delivery_report_view.sql

```diff
@@ -50,23 +50,23 @@ COALESCE(agg_sdr.gm_0_3,0) as gm_0_3,
 COALESCE(agg_sdr.gm_3_5,0) as gm_3_5,
 COALESCE(agg_sdr.children_0_3,0) as children_0_3,
 COALESCE(agg_sdr.children_3_5,0) as children_3_5
-FROM "public"."awc_location_months_local" "awc_location_months"
+FROM "public"."awc_location_local"
 LEFT join agg_awc on (
-        ("agg_awc"."month" = "awc_location_months"."month") AND
-        ("agg_awc"."state_id" = "awc_location_months"."state_id") AND
-        ("agg_awc"."district_id" = "awc_location_months"."district_id") AND
-        ("agg_awc"."block_id" = "awc_location_months"."block_id") AND
-        ("agg_awc"."supervisor_id" = "awc_location_months"."supervisor_id") AND
-        ("agg_awc"."aggregation_level" = "awc_location_months"."aggregation_level") AND
-        ("agg_awc"."awc_id" = "awc_location_months"."awc_id")
+        ("agg_awc"."state_id" = "awc_location_local"."state_id") AND
+        ("agg_awc"."district_id" = "awc_location_local"."district_id") AND
+        ("agg_awc"."block_id" = "awc_location_local"."block_id") AND
+        ("agg_awc"."supervisor_id" = "awc_location_local"."supervisor_id") AND
+        ("agg_awc"."aggregation_level" = "awc_location_local"."aggregation_level") AND
+        ("agg_awc"."awc_id" = "awc_location_local"."doc_id")
 )
 LEFT JOIN agg_service_delivery_report agg_sdr on (
-        ("agg_sdr"."month" = "awc_location_months"."month") AND
-        ("agg_sdr"."state_id" = "awc_location_months"."state_id") AND
-        ("agg_sdr"."district_id" = "awc_location_months"."district_id") AND
-        ("agg_sdr"."block_id" = "awc_location_months"."block_id") AND
-        ("agg_sdr"."supervisor_id" = "awc_location_months"."supervisor_id") AND
-        ("agg_sdr"."aggregation_level" = "awc_location_months"."aggregation_level") AND
-        ("agg_sdr"."awc_id" = "awc_location_months"."awc_id")
+        ("agg_sdr"."month" = "agg_awc"."month") AND
+        ("agg_sdr"."state_id" = "awc_location_local"."state_id") AND
+        ("agg_sdr"."district_id" = "awc_location_local"."district_id") AND
+        ("agg_sdr"."block_id" = "awc_location_local"."block_id") AND
+        ("agg_sdr"."supervisor_id" = "awc_location_local"."supervisor_id") AND
+        ("agg_sdr"."aggregation_level" = "awc_location_local"."aggregation_level") AND
+        ("agg_sdr"."awc_id" = "awc_location_local"."doc_id")
 )
 
+WHERE awc_location_local.state_is_test IS DISTINCT FROM 1
```

Comment:
this is the main change, yeah? what's with the other stuff?

Reply:
I think most of our reports do this filtering at the query level rather than in the view itself (that way if you want test data you can get it). It might be better to follow that pattern here.

Reply:
Sure. yeah I think that makes sense